source/simulator/solver.cc

```diff
@@ -790,10 +791,15 @@ namespace aspect
                 // output solver history
                 std::ofstream f((parameters.output_directory+"solver_history.txt").c_str());
 
-                for (unsigned int i=0; i<solver_control_cheap.get_history_data().size(); ++i)
-                  f << i << " " << solver_control_cheap.get_history_data()[i] << "\n";
+                if (parameters.n_cheap_stokes_solver_steps > 0)
```

Comment:
Without the if statement I got the following error message when setting the number of cheap solver steps to 0 and the number of expensive solver steps to (e.g.) 10:
"The SolverControl object was asked for the solver history data, but there is no data. Possibly you requested the data before the solver was run." 
With the if statement I don't run into this issue and get the solver history without any values from the cheap solver steps.